### PR TITLE
[Access] Add support for programs cache on access nodes - v0.33

### PR DESCRIPTION
**Pull request by peterargue**
Closes: https://github.com/onflow/flow-go/issues/5278

Add support for running the programs cache on Access nodes with script execution enabled. This improves overall script execution performance since the most common cadence contracts will be cached.

### CODE REVIEW

**Comment by janezpodhostnik** [comment]
@peterargue is this ready for review?

**Comment by peterargue** [comment]
> @peterargue is this ready for review?

yes, it should be ready now

**Review by janezpodhostnik — COMMENTED** [review]
Nice work!

File: utils/dsl/dsl.go [hunk 1]
```diff
@@ -69,8 +69,9 @@ func (i Import) ToCadence() string {
 }
 
 type UpdateAccountCode struct {
```
[the remark this answers is not in the document]
Comment (peterargue) [reply utils/dsl/dsl.go]: [058800e](https://github.com/onflow/flow-go/pull/5585/commits/058800ef5e9263cc50ce90f6009e9e3a531ee0fe)

File: fvm/context.go [hunk 2]
```diff
@@ -375,3 +379,11 @@ func WithEVMEnabled(enabled bool) Option {
 		return ctx
 	}
 }
+
+// WithAllowProgramCacheWritesInScriptsEnabled enables caching of programs accessed by scripts
+func WithAllowProgramCacheWritesInScriptsEnabled(enabled bool) Option {
```
[the remark this answers is not in the document]
Comment (peterargue) [reply fvm/context.go:384]: [a169120](https://github.com/onflow/flow-go/pull/5585/commits/a1691207c1b86750ce9e0664dc2f15169a85787d)

**Review by janezpodhostnik — APPROVED** [review]

**Review by zhangchiqing — APPROVED** [review]

**Pull request merged**